packages/components/avatar/src/avatar.ts

```diff
@@ -1,89 +1,60 @@
-function useImageLoadingStatus(src?: string) {
-  const loadingStatus = ref<ImageLoadingStatus>('idle')
-
-  onMounted(() => {
-    if (!src) {
-      loadingStatus.value = 'error'
-      return
-    }
-
-    let isMounted = true
-    const image = new window.Image()
-
-    const updateStatus = (status: ImageLoadingStatus) => () => {
-      if (!isMounted)
-        return
-      loadingStatus.value = status
-    }
-
-    loadingStatus.value = 'loading'
-    image.onload = updateStatus('loaded')
-    image.onerror = updateStatus('error')
-    image.src = src
-
-    onUnmounted(() => {
-      isMounted = false
-    })
-  })
-
-  return loadingStatus
-}
+type ImageLoadingStatus = 'idle' | 'loading' | 'loaded' | 'error'
 
 /* -------------------------------------------------------------------------------------------------
  * Avatar
  * ----------------------------------------------------------------------------------------------- */
 
 const AVATAR_NAME = 'Avatar'
-type ScopedProps<P> = P & { __scopeAvatar?: Scope }
-const [createAvatarProvide, createAvatarScope] = createProvideScope(AVATAR_NAME)
-
-type ImageLoadingStatus = 'idle' | 'loading' | 'loaded' | 'error'
 
 type AvatarProvideValue = {
-  imageLoadingStatus: ImageLoadingStatus
-  onImageLoadingStatusChange(status: ImageLoadingStatus): void
+  src?: string
+  loadingStatus: ImageLoadingStatus
 }
+const PROVIDER_KEY = Symbol(AVATAR_NAME) as InjectionKey<AvatarProvideValue>
 
-const [AvatarProvider, useAvatarInject] = createAvatarProvide<AvatarProvideValue>(AVATAR_NAME)
+const [AvatarProvider, useAvatarInject] = createProvide<AvatarProvideValue>(PROVIDER_KEY, AVATAR_NAME)
 
-type AvatarElement = ComponentPropsWithoutRef<typeof Primitive.span>
+// type AvatarElement = ComponentPropsWithoutRef<typeof Primitive.span>
 type PrimitiveSpanProps = ComponentPropsWithoutRef<typeof Primitive.span>
-interface AvatarProps extends PrimitiveSpanProps {}
+interface AvatarProps extends PrimitiveSpanProps {
+  src?: string
+}
+
+/* -------------------------------------------------------------------------------------------------
+ * Avatar
+ * ----------------------------------------------------------------------------------------------- */
 
-const Avatar = defineComponent<ScopedProps<AvatarProps>>({
+const Avatar = defineComponent<AvatarProps>({
   name: AVATAR_NAME,
-  inheritAttrs: false,
-  setup(props, { attrs, slots, emit }) {
-    const forwardedRef = ref<AvatarElement>()
+  setup(_, { attrs, slots, expose }) {
+    const innerRef = ref<typeof Primitive.span | null>(null)
 
-    onMounted(() => {
-      emit('update:ref', forwardedRef.value)
-    })
-    const { __scopeAvatar, ...avatarProps } = attrs as any
-    const imageLoadingStatus = ref<ImageLoadingStatus>('idle')
-
-    AvatarProvider({
-      scope: __scopeAvatar,
-      imageLoadingStatus: imageLoadingStatus.value,
-      onImageLoadingStatusChange: (status: ImageLoadingStatus) => {
-        imageLoadingStatus.value = status
-      },
-    })
```

Comment:
This structure and approach is correct because images are being loaded on the `AvatarImage`  side, you have downloaded the structure to a custom structure, you should think more globally. you also need to check the scope structures in the radix.